src/paths/widest_paths.c

```diff
@@ -0,0 +1,745 @@
+ *    Each row contains the widths from a single source, to the
+ *    vertices given in the \c to argument.
+ *    Unreachable vertices have width \c IGRAPH_NEGINFINITY, and vertices
+ *    have a width of \c IGRAPH_POSINFINITY to themselves.
+ * \param from The source vertices.
+ * \param to The target vertices. It is not allowed to include a
+ *    vertex twice or more.
+ * \param weights The edge weights. Edge weights can be negative. If this
+ *        is a null pointer or if any edge weight is NaN, then an error
+ *        is returned.
+ * \param mode For directed graphs; whether to follow paths along edge
+ *    directions (\c IGRAPH_OUT), or the opposite (\c IGRAPH_IN), or
+ *    ignore edge directions completely (\c IGRAPH_ALL). It is ignored
+ *    for undirected graphs.
+ * \return Error code.
+ *
+ * Time complexity: O(s*|E|log|E|+|V|), where |V| is the number of
+ * vertices, |E| the number of edges and s the number of sources.
+ *
+ * \sa \ref igraph_widest_paths_floyd_warshall() for a variant that runs faster
+ * on dense graphs.
+ */
+igraph_error_t igraph_widest_paths_dijkstra(const igraph_t *graph,
+                                   igraph_matrix_t *res,
+                                   const igraph_vs_t from,
+                                   const igraph_vs_t to,
+                                   const igraph_vector_t *weights,
+                                   igraph_neimode_t mode) {
+
+    /* Implementation details: This is a Dijkstra algorithm with a
+    binary heap, modified to support widest paths. The heap is indexed,
+    so it stores both the widest path to a node, as well as it's index. We
+    use a 2 way heap so that we can query indexes directly in the heap.
+
+    To adapt a Dijkstra to handle widest path, instead of prioritising candidate
+    nodes with the minimum distance, we prioritise those with the maximum
+    width instead. When adding a node into our set of 'completed' nodes, we
+    update all neighbouring nodes with a width that is equal to the min of the
+    width to the current node and the width of the edge.
+
+    We denote the widest path from a node to itself as infinity, and the widest
+    path from a node to a node it cannot reach as negative infinity.
+    */
+
+    igraph_integer_t no_of_nodes = igraph_vcount(graph);
+    igraph_integer_t no_of_edges = igraph_ecount(graph);
+    igraph_2wheap_t Q;
+    igraph_vit_t fromvit, tovit;
+    igraph_integer_t no_of_from, no_of_to;
+    igraph_lazy_inclist_t inclist;
+    igraph_integer_t i, j;
+    igraph_real_t my_posinfinity = IGRAPH_POSINFINITY;
+    igraph_real_t my_neginfinity = IGRAPH_NEGINFINITY;
+    igraph_bool_t all_to;
+    igraph_vector_int_t indexv;
+
+    if (!weights) {
+        IGRAPH_ERROR("Weight vector is required.", IGRAPH_EINVAL);
+    }
+
+    if (igraph_vector_size(weights) != no_of_edges) {
+        IGRAPH_ERRORF("Weight vector length (%ld) does not match number "
```

Comment:
This is the cause of the test failure. Instead of `"...%ld..."`, use `"...%"IGRAPH_PRId"..."`

Reply:
Ah thanks for that. I was able to compile successfully locally, and forgot to check the Github tests after pushing. 

Out of curiosity, is there a way to run the tests locally before pushing publically to the PR? Sometimes I get worried that people will start reviewing before I have a chance to fix whatever bugs come up in the test.